README.md

#The code repo for the Earth 2 Mars robots for @Bristol 

Instructions for getting code working on all robots.

##1 Download Repo

First delete the old repo and pibot folder:

```
sudo rm -rf ~/v2 ~/pibot 
```
Now install the new repo:
```
git clone https://github.com/pi-bot/.v2
```


##2 Update the firmware
This pushes the fixed code to the microcontroller:
```
cd ~/.v2/arduino-firmware 
./upload.sh
```
Messages onscreen will show that this is successfull

##3 Install the python modules
OK now we can place the python code deep into the system:

```
sudo mkdir /usr/lib/python2.7/pibot
sudo cp ~/.v2/python-code/*  /usr/lib/python2.7/pibot/
```
Now we must call modules in all python using a **pibot.modulename** convention.

##4 Use the new movement functions

The new movement functions are demonstrated in a new test:

```
cd ~/.v2/tests
python NewMattTest.py
```
All documents and test codes can now be updated! 

Any questions/comments email **harry@pibot.org**

See License for details of copying and using the code.


wget https://github.com/pi-bot/.v2/blob/master//core-tests/stepper.py
wget https://github.com/pi-bot/.v2/blob/master//core-tests/stepper2.py




newTests/history.txt
    1  sudo apt-get update
    2  sudo apt-get dist-upgrade
    3  sudo apt-get update
    4  sudo apt-get upgrade
    5  sudo shutdown
    6  history
    7  sudo apt-get install python-pip
    8  sudo apt-get install git
    9  sudo apt-get install libi2c-dev i2c-tools minicom python-dev
   10  sudo apt-get install libi2c-dev i2c-tools python-dev
   11  sudo apt-get install libi2c-dev  minicom python-dev
   12  sudo apt-get install libi2c-dev  python-dev
   13  sudo apt-get install libi2c-dev  minicom python-dev
   14  `uname -m
   15  uname -m
   16  cat /proc/cpuinfo
   17  kernel_version=`uname -a|awk '{print $3}'`
   18  if [ $kernel_version == "4.4" ]; then     echo "version is 4.4"; fi
   19  uname -a|awk '{print $3}'
   20  cat /proc/cmdline | awk -v RS=" " -F= '/boardrev/ { print $2 }'
   21  cat /proc/cmdline 
   22  cat /proc/cpuinfo | grep 'Revision' | awk '{print $3}' | sed 's/^1000//'
   23  ==clear
   24  clear
   25  cat /boot/config.txt
   26  cat /boot/cmdline.txt
   27  cat /boot/config.txt
   28  cat /boot/cmdline.txt
   29  sed -man
   30  cat /boot/cmdline.txt
   31  sudo sed -e "s/console=serial0,115200//g" /boot/cmdline.txt
   32  cat /boot/cmdline.txt
   33  sudo   sed -e "s/console=serial0,115200//g" -i /boot/cmdline.txt
   34  cat /boot/cmdline.txt
   35  sudo ln -s /usr/bin/avrdude-autoreset /usr/bin/avrdude
   36  cat /usr/bin/autoreset
   37  sudo   sed -e "s/pin = 22/pin = 7/g" -i /usr/bin/autoreset
   38  sudo   sed -e "s/time.sleep(0.12)/time.sleep(0.32)/g" -i /usr/bin/autoreset
   39  cat /usr/bin/autoreset
   40  cd /usr/
   41  ls
   42  cd bin
   43  ls
   44  cd /usr/share/arduino/hardware/arduino
   45  ls
   46  sudo vim boards.txt 
   47  cd /tmp
   48  sudo   mv boards.txt board.txt.bak
   49  sudo   mv /tmp/boards.txt .
   50  sudo   mv programmers.txt programmers.txt.bak
   51  sudo   mv /tmp/programmers.txt .
   52  vim test1.sh
   53  chom -x test1.sh 
   54  chmod  -x test1.sh 
   55  .. test1.sh 
   56  bash test1.sh 
   57  vim test1.sh
   58  bash test1.sh 
   59  vim test1.sh
   60  bash test1.sh 
   61  vim test1.sh
   62  bash test1.sh 
   63  vim test1.sh
   64  bash test1.sh 
   65  x!
   66  vim test1.sh
   67  bash test1.sh 
   68  vim test1.sh
   69  bash test1.sh 
   70  vim test1.sh
   71  bash test1.sh 
   72  vim test1.sh
   73  bash test1.sh 
   74  vim test1.sh
   75  bash test1.sh 
   76  vim test1.sh
   77  bash test1.sh 
   78  vim test1.sh
   79  bash test1.sh 
   80  vim test1.sh
   81  bash test1.sh 
   82  vim test1.sh
   83  bash test1.sh 
   84  vim test1.sh
   85  bash test1.sh 
   86  vim test1.sh
   87  bash test1.sh 
   88  vim test1.sh
   89  sudo bash test1.sh 
   90  vim test1.sh
   91  sudo bash test1.sh 
   92  ls
   93  cd tempPiBot/
   94  ls
   95  cd avrdude-rpi/
   96  ls
   97  cd ../../
   98  ls
   99  vim test1.sh 
  100  sudo reboot 
  101  cd /usr/share/arduino/hardware/arduino
  102  ls
  103  ls -la
  104  sudo chown root:root boards.txt 
  105  sudo chown root:root programmers.txt 
  106  sudo apt-get remove arduino
  107  sudo apt-get install arduino
  108  ls
  109  cat test1.sh 
  110  vim  test1.sh 
  111  ls
  112  vim boards.txt
  113  sudo vim boards.txt
  114  sudo vim programmers.txt
  115  cs ~
  116  cd ~'
  117  cd ~
  118  ls
  119  git clone https://github.com/pi-bot/.v2
  120  cd .v2
  121  ls
  122  cd arduino-firmware/
  123  la
  124  ls
  125  ./ upload.sh
  126  bash ./upload.sh 
  127  sudo bash ./upload.sh 
  128  cd /usr/bin/
  129  ls
  130  cin avrdude-original 
  131  vim  avrdude-original 
  132  vim  avrdude-autoreset 
  133  vim  autoreset 
  134  sudo vim /boot/config.txt
  135  sudo vim /boot/cmdline.txt
  136  vim  autoreset 
  137  cd /usr/share/arduino/
  138  ls
  139  cd hardware/
  140  ls
  141  cd arduino/
  142  ls
  143  vim programmers.txt.bak
  144  sudo mv programmers.txt v2.txt
  145  sudo mv programmers.txt.bak programmers.txt
  146  ls
  147  cd /etc/udev/
  148  ls
  149  cd rules.d
  150  ls
  151  csd .v2
  152  cd .v2
  153  cd arduino-firmware/
  154  ls
  155  sudo ./ upload.sh
  156  sudo bash ./ upload.sh
  157  sudo bash upload.sh
  158  cd .v2
  159  ls
  160  cd arduino-firmware/
  161  ls
  162  sudo ./upload.sh
  163  cd /etc/
  164  vim avrdude.conf
  165  ls
  166  cd /usr/share/arduino
  167  ls
  168  cd hardware
  169  ls
  170  cd arduino/
  171  ls
  172  cd~
  173  cd ~
  174  ls
  175  cd .v2
  176  ls
  177  cd /usr/bin
  178  ls
  179  sudo vim avrdude-autoreset
  180  sudo vim autoreset
  181  cd .v2/arduino-firmware/
  182  ls
  183  sudo vim upload.sh
  184  ./upload.sh
  185  sudo ./upload.sh
  186  cd .v2
  187  cd arduino-firmware/
  188  ls
  189  cd ../
  190  cd core-tests/


install.sh
#! /bin/bash
echo -e "\E[32m"
echo " ____  __  ____   __  ____      ____  ____  ____  _  _  ____  ";
echo "(  _ \(  )(  _ \ /  \(_  _)    / ___)(  __)(_  _)/ )( \(  _ \ ";
echo " ) __/ )(  ) _ ((  O ) )(      \___ \ ) _)   )(  ) \/ ( ) __/ ";
echo "(__)  (__)(____/ \__/ (__)     (____/(____) (__) \____/(__)   ";
echo -e "\E[0m"
echo "Welcome to the PiBot Installer."
echo " "
echo "Requirements:"
echo "1) Must be connected to the internet"
echo "2) This script must be run as root user"
echo " "
echo "Assumptions:"
echo "1) It is assummed that the following dependencies are included in your Raspian Distro:"
echo "   - python-pip       alternative Python package installer"
echo "   - git              fast, scalable, distributed revision control system"
echo "   - libi2c-dev       userspace I2C programming library development files"
echo "   - python-serial    pyserial - module encapsulating access for the serial port"
echo "   - python-rpi.gpio  Python GPIO module for Raspberry Pi"
echo "   - i2c-tools        This Python module allows SMBus access through the I2C /dev"
echo "   - python-smbus     Python bindings for Linux SMBus access through i2c-dev"
echo "   - python3-smbus    Python3 bindings for Linux SMBus access through i2c-dev"
echo "REQUIRED PACKAGES:"
echo "1) The following dependencies will be downloaded as part of this install:"
echo "   - arduino          AVR development board IDE and built-in libraries"
echo "   - minicom          friendly menu driven serial communication program"
echo "2) Clone, build wiringPi in GrovePi/Script and install it"
echo "3) Removes I2C and SPI from modprobe blacklist /etc/modprobe.d/raspi-blacklist.conf"
echo "4) Adds I2C-dev, i2c-bcm2708 and spi-dev to /etc/modules"
echo "5) Installs gertboard avrdude_5.10-4_armhf.deb package"
echo "6) Runs gertboard setup"
echo "   - configures avrdude"
echo "   - downloads gertboard known boards and programmers"
echo "   - replaces avrsetup with gertboards version"
echo "   - in /etc/inittab comments out lines containing AMA0"
echo "   - in /boot/cmdline.txt removes: console=ttyAMA0,115200 kgdboc=ttyAMA0,115200 console=tty1"
echo "   - in /usr/share/arduino/hardware/arduino creates backup of boards.txt"
echo "   - in /usr/share/arduino/hardware/arduino creates backup of programmers.txt"
echo " "
echo "Special thanks to Joe Sanford at Tufts University. This script was derived from his work. Thank you Joe!"
echo " "
echo "Raspberry Pi will reboot after completion."
echo " "
echo " "
sleep 5

echo " "
echo "Check for internet connectivity..."
echo "=================================="
wget -q --tries=2 --timeout=100 http://google.com -O /dev/null
if [ $? -eq 0 ];then
        echo "Connected"
else
        echo "Unable to Connect, try again !!!"
        exit 0
fi

USER_ID=$(/usr/bin/id -u)
USER_NAME=$(/usr/bin/who am i | awk '{ print $1 }')
SCRIPT_PATH=$(/usr/bin/realpath $0)
DIR_PATH=$(/usr/bin/dirname ${SCRIPT_PATH} | sed 's/\/Script$//')

if [ ${USER_ID} -ne 0 ]; then
    echo "Please run this as root."
    exit 1
fi

echo " "
echo "Installing Dependencies"
echo "======================="
sudo apt-get install libi2c-dev arduino minicom python-dev -y
sudo apt-get purge python-rpi.gpio -y
sudo apt-get purge python3-rpi.gpio -y
sudo apt-get install python-rpi.gpio -y
sudo apt-get install python3-rpi.gpio -y
sudo pip install -U RPi.GPIO
echo "Dependencies installed"

if [ -d wiringPi ]; then
    cd wiringPi
    git pull
else
    git clone git://git.drogon.net/wiringPi
    cd wiringPi
fi

./build
RES=$?

if [ $RES -ne 0 ]; then
  echo "Something went wrong building/installing wiringPi, exiting."
  exit 1
fi

echo "wiringPi Installed"

echo " "
RASPI_BL="/etc/modprobe.d/raspi-blacklist.conf.bak"
MODS="i2c spi"
if [ -f ${RASPI_BL} ]; then
    echo "Removing blacklist from ${RASPI_BL} . . ."
    echo "=================================================================="
    echo " "
    for i in ${MODS}
    do
        MOD_NAME=$(echo $i | tr [a-z] [A-Z])
        sudo sed -i -e "s/blacklist ${i}-bcm2708/#blacklist ${i}-bcm2708/g" ${RASPI_BL}
        echo "${MOD_NAME} not present or removed from blacklist"
    done
fi

#Adding in /etc/modules
echo " "
echo "Adding I2C-dev and SPI-dev in /etc/modules . . ."
echo "================================================"
if grep -q "i2c-dev" /etc/modules; then
        echo "I2C-dev already present"
else
        echo i2c-dev >> /etc/modules
        echo "I2C-dev added"
fi
if grep -q "i2c-bcm2708" /etc/modules; then
        echo "i2c-bcm2708 already present"
else
        echo i2c-bcm2708 >> /etc/modules
        echo "i2c-bcm2708 added"
fi
if grep -q "spi-dev" /etc/modules; then
        echo "spi-dev already present"
else
        echo spi-dev >> /etc/modules
        echo "spi-dev added"
fi

echo " "
echo "Making I2C changes in /boot/config.txt . . ."
echo "================================================"

BOOT_CONFIG="/boot/config.txt"
DTPARAMS="i2c1 i2c_arm"
for i in ${DTPARAMS}
do
    if grep -q "^dtparam=${i}=on$" ${BOOT_CONFIG}; then
        echo "${i} already present"
    else
        echo "dtparam=${i}=on" >> /boot/config.txt
    fi
done

sudo adduser ${USER_NAME} i2c
sudo chmod +x ${DIR_PATH}/Software/Scratch/GrovePi_Scratch_Scripts/*.sh

#Adding ARDUINO setup files
echo " "
echo "Making changes to Arduino . . ."
echo "==============================="

cd /tmp
wget http://project-downloads.drogon.net/gertboard/avrdude_5.10-4_armhf.deb
sudo dpkg -i avrdude_5.10-4_armhf.deb
sudo chmod 4755 /usr/bin/avrdude

cd /tmp
wget http://project-downloads.drogon.net/gertboard/setup.sh
chmod +x setup.sh
sudo ./setup.sh
echo " "
echo "If you see errors related to /etc/inittab, it's fine."
echo "/etc/inittab has been deprecated in favor of systemd,"
echo "cfr. https://www.raspberrypi.org/forums/viewtopic.php?f=66&t=123081"

echo " "
echo "Install smbus for python"
sudo apt-get install python-smbus -y

echo " "
echo "Making libraries global . . ."
echo "============================="
if [ -d /usr/lib/python2.7/dist-packages ]; then
    sudo cp ${DIR_PATH}/Script/grove.pth /usr/lib/python2.7/dist-packages/grove.pth
else
    echo "/usr/lib/python2.7/dist-packages not found, exiting"
    exit 1
fi

echo " "
echo "Please restart to implement changes!"
echo "  _____  ______  _____ _______       _____ _______ "
echo " |  __ \|  ____|/ ____|__   __|/\   |  __ \__   __|"
echo " | |__) | |__  | (___    | |  /  \  | |__) | | |   "
echo " |  _  /|  __|  \___ \   | | / /\ \ |  _  /  | |   "
echo " | | \ \| |____ ____) |  | |/ ____ \| | \ \  | |   "
echo " |_|  \_\______|_____/   |_/_/    \_\_|  \_\ |_|   "
echo " "
echo "Please restart to implement changes!"
echo "To Restart type sudo reboot"

echo "To finish changes, we will reboot the Pi."
echo "Pi must reboot for changes and updates to take effect."
echo "If you need to abort the reboot, press Ctrl+C.  Otherwise, reboot!"
echo "Rebooting in 5 seconds!"

Get Kernal Version 

kernel_version=`uname -a|awk '{print $3}'`
if [ $kernel_version == "4.4" ]
then
    echo "version is 4.4"
fi

Get Hardware Version:
cat /proc/cmdline | awk -v RS=" " -F= '/boardrev/ { print $2 }'

Revision : a21041
REVISION="(cat /proc/cpuinfo | grep 'Revision' | awk '{print $3}' | sed 's/^1000//')"
SERIAL="$(cat /proc/cpuinfo | grep Serial | cut -d ':' -f 2)"
echo $SERIAL
if [ "$REVISION" = "a02082" ]
then
  echo "Board is Raspberry Pi version 3. made by sony in the UK (Newport)"
  echo "Serial number is : $SERIAL"
else if [ "$REVISION" = "a22082" ]
  echo "Board is Raspberry Pi version 3. made in China"
else 
  echo "Board is not a Raspberry Pi 3"
  echo "Serial number is :" $SERIAL
fi
Check if the script is run as root:


if [ "$EUID" -ne 0 ]
  then echo "Please run as root"
  exit
fi


